### PR TITLE
HPCC-22322 Propagate published rowTag and allow client to specify

### PR DESCRIPTION
**Pull request by jakesmith**
XML and JSON published row tags were not being passed through
to dafilesrv, and there was no mechanism to allow the client to
override the row tag to be used if they needed (e.g. if there
is no published row tag).

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@rpastrana - please review

File: fs/dafsserver/dafsserver.cpp [hunk 1]
```diff
@@ -1882,6 +1884,11 @@ class CRemoteXmlReadActivity : public CRemoteMarkupReadActivity
 public:
     CRemoteXmlReadActivity(IPropertyTree &config, IFileDescriptor *fileDesc) : PARENT(config, fileDesc, TAKxmlread)
     {
+        xpath.set("/Dataset/");
+        if (customRowTag.isEmpty()) // no override
+            fileDesc->queryProperties().getProp("@rowTag", xpath);
```
Comment (rpastrana) [inline fs/dafsserver/dafsserver.cpp]: does getProp append on to xpath? if it performs a set(), then we have slightly different behavior if we use this.customRowTag, and properties[@rowTag]

**Review by rpastrana — APPROVED** [review]
@jakesmith looks good, just one question.

File: fs/dafsserver/dafsserver.cpp (hunk 1 above)
Comment (jakesmith) [reply fs/dafsserver/dafsserver.cpp]: @rpastrana IPropertyTree::getProp (and other get function) append to the output StringBuffer.

**Comment by jakesmith** [comment]
@richardkchapman - ready to merge.

**Pull request merged**